crates/egui/src/context.rs

```diff
@@ -2071,9 +2071,14 @@ impl Context {
     /// Move all the graphics at the given layer.
     ///
     /// Can be used to implement drag-and-drop (see relevant demo).
-    pub fn translate_layer(&self, layer_id: LayerId, delta: Vec2) {
+    pub fn transform_layer(&self, layer_id: LayerId, delta: Vec2, scale: f32) {
```

Comment:
This method used to be only an immediate thing, applying a transform to shapes created this frame.
Now it _also_ stores the transform in memory, applying it to all interaction, but _not_ to future paint shapes. This is not great.

We need to reconsider this API.

I think it would be ideal if we call it `set_layer_transform` instead. It can store the transform in memory, and then we apply the transform to the shapes during `GraphicLayers::drain`.